src/coreclr/minipal/Windows/doublemapping.cpp

```diff
@@ -0,0 +1,246 @@
+                        FILE_MAP_EXECUTE | FILE_MAP_READ | FILE_MAP_WRITE,
+                        HIDWORD((int64_t)offset),
+                        LODWORD((int64_t)offset),
+                        size);
+    }
+
+    // We will do one scan from [pMinAddr .. pMaxAddr]
+    // First align the tryAddr up to next 64k base address.
+    // See docs for VirtualAllocEx and lpAddress and 64k alignment for reasons.
+    //
+    BYTE *   tryAddr            = (BYTE *)ALIGN_UP((BYTE *)pMinAddr, VIRTUAL_ALLOC_RESERVE_GRANULARITY);
+    bool     virtualQueryFailed = false;
+    bool     faultInjected      = false;
+    unsigned virtualQueryCount  = 0;
+
+    // Now scan memory and try to find a free block of the size requested.
+    while ((tryAddr + size) <= (BYTE *) pMaxAddr)
+    {
+        MEMORY_BASIC_INFORMATION mbInfo;
+
+        // Use VirtualQuery to find out if this address is MEM_FREE
+        //
+        virtualQueryCount++;
+        if (!VirtualQuery((LPCVOID)tryAddr, &mbInfo, sizeof(mbInfo)))
+        {
+            // Exit and return nullptr if the VirtualQuery call fails.
+            virtualQueryFailed = true;
+            break;
+        }
+
+        // Is there enough memory free from this start location?
+        // Note that for most versions of UNIX the mbInfo.RegionSize returned will always be 0
+        if ((mbInfo.State == MEM_FREE) &&
+            (mbInfo.RegionSize >= (SIZE_T) size || mbInfo.RegionSize == 0))
+        {
+            // Try reserving the memory using VirtualAlloc now
+            pResult = (BYTE*)MapViewOfFileEx((HANDLE)mapperHandle,
+                        FILE_MAP_EXECUTE | FILE_MAP_READ | FILE_MAP_WRITE,
+                        HIDWORD((int64_t)offset),
+                        LODWORD((int64_t)offset),
+                        size,
+                        tryAddr);
+
+            // Normally this will be successful
+            //
+            if (pResult != nullptr)
+            {
+                // return pResult
+                break;
+            }
+
+#ifdef _DEBUG
+            // if (ShouldInjectFaultInRange())
+            // {
+            //     // return nullptr (failure)
+            //     faultInjected = true;
+            //     break;
+            // }
+#endif // _DEBUG
+
+            // On UNIX we can also fail if our request size 'dwSize' is larger than 64K and
+            // and our tryAddr is pointing at a small MEM_FREE region (smaller than 'dwSize')
+            // However we can't distinguish between this and the race case.
+
+            // We might fail in a race.  So just move on to next region and continue trying
+            tryAddr = tryAddr + VIRTUAL_ALLOC_RESERVE_GRANULARITY;
+        }
+        else
+        {
+            // Try another section of memory
+            tryAddr = max(tryAddr + VIRTUAL_ALLOC_RESERVE_GRANULARITY,
+                          (BYTE*) mbInfo.BaseAddress + mbInfo.RegionSize);
+        }
+    }
+
+    // STRESS_LOG7(LF_JIT, LL_INFO100,
```

Comment:
@jkotas using stresslog from here would require some clunky stuff. I don't remember ever needing this info from the stresslog in the past years. I was going to remove these, but I wanted to know your opinion first.

Reply:
Sounds good to me.